### DISCUSSION TITLE
Svelte: Times New Roman Font Not Applying to Mathematical Expressions in OpenWebUI

### DISCUSSION THREAD

**Discussion by currytut** [open-webui/open-webui#5796] [General]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by currytut** [comment]
Problem Summary:
I'm customizing the OpenWebUI project, which uses Svelte for the frontend, and I want to display mathematical expressions (numbers, variables, and operators) in Times New Roman font. However, despite making changes to the CSS and Svelte components, the font is not applying correctly to the mathematical content.

What I've Tried:
CSS:

I’ve added a .math-content CSS class with font-family: 'Times New Roman', Times, serif; to ensure the font applies to all mathematical expressions. The class is intended to be used for numbers, variables, and operators in the LLM's responses.
Svelte Modifications:

In the relevant Svelte components (like MessageInput.svelte and Messages.svelte), I used JavaScript to wrap mathematical expressions in <span> tags with the .math-content class using a regular expression.
The formattedMessage logic is applied using innerHTML, but the .math-content class is not being consistently applied to the expressions.
Font Installation:

I’ve ensured that Times New Roman is installed on my system and also added a Google Fonts fallback link, but the math expressions still don’t display in the correct font.
Goal:
My goal is for all mathematical expressions (numbers, variables like x, and operators such as +, -, *, /, ^) in the LLM's responses to be displayed in Times New Roman font. Currently, the font is not rendering properly for the math expressions, and I’m unsure why the .math-content class isn’t being applied as expected.

**Comment by currytut** [comment]
Questions or Additional Context:
Why isn't the .math-content class applying to the mathematical expressions in the LLM's responses, despite adding it to the relevant CSS and Svelte components?
Could there be an issue with how Svelte renders the innerHTML when using dynamically generated content with <span> tags?
Are there any known issues with CSS specificity or font inheritance that could be causing the Times New Roman font not to render as expected?
Could Docker or any other part of my setup (Kali Linux, OpenWebUI) be interfering with the proper application of CSS or font rendering?
What steps should I take to debug or inspect whether the CSS or Svelte logic is being overridden or misapplied?

**Comment by sp301415** [comment]
I'm afraid it will be quite difficult. Open-WebUI currently uses KaTeX to render math contents, and according to https://github.com/KaTeX/KaTeX/discussions/3716, changing fonts in KaTeX seems to be non-trivial.
